integrations/sharepoint/src/SharepointClient.ts

```diff
@@ -0,0 +1,441 @@
+          expirationDateTime: new Date(Date.now() + 1000 * 60 * 60 * 24 * 30).toISOString(),
+        },
+        {
+          headers: {
+            Authorization: `Bearer ${token.accessToken}`,
+            Accept: "application/json;odata=verbose",
+          },
+        }
+      )
+      .catch(handleAxiosError);
+
+    if (!res) {
+      throw new sdk.RuntimeError(`Error registering webhook`);
+    }
+
+    const webhookId = res.data.d.id;
+
+    return webhookId;
+  }
+
+  /**
+   * Unregisters a webhook for the configured document library.
+   * @param webhookId - The ID of the webhook to unregister.
+   */
+  async unregisterWebhook(webhookId: string): Promise<void> {
+    const listId = this.getDocumentLibraryListId();
+    const url = `https://${this.primaryDomain}.sharepoint.com/sites/${this.siteName}/_api/web/lists('${listId}')/subscriptions('${webhookId}')`;
+    const token = await this.acquireToken();
+    await axios
+      .delete(url, {
+        headers: {
+          Authorization: `Bearer ${token.accessToken}`,
+          Accept: "application/json;odata=verbose",
+        },
+      })
+      .catch(handleAxiosError);
+  }
+
+  /**
+   * Get the list of documents from the configured document library.
+   * @returns - The list of documents.
+   */
+  private async listItems(): Promise<SharePointItem[]> {
+    const token = await this.acquireToken();
+    const url = `https://${this.primaryDomain}.sharepoint.com/sites/${this.siteName}/_api/web/lists/getbytitle('${this.documentLibraryName}')/items`;
+    const res = await axios.get<SharePointItemsResponse>(url, {
+      headers: {
+        Authorization: `Bearer ${token.accessToken}`,
+        Accept: "application/json;odata=verbose",
+      },
+    });
+    return res.data.d.results;
+  }
+
+  /**
+   * Get the file name from the SharePoint list item
+   * @param listItemIndex - The item index to get the file name for
+   * @returns - The file name or null if the file does not exist
+   */
+  private async getFileName(listItemIndex: number): Promise<string | null> {
+    // sample url -https://botpressio836.sharepoint.com/sites/DemoStandardTeamPage/_api/web/lists/getbytitle('NewDL')/items(3)/File
+    const url = `https://${this.primaryDomain}.sharepoint.com/sites/${this.siteName}/_api/web/lists/getbytitle('${this.documentLibraryName}')/items(${listItemIndex})/File`;
+    const token = await this.acquireToken();
+    const res = await axios
+      .get(url, {
+        headers: {
+          Authorization: `Bearer ${token.accessToken}`,
+          Accept: "application/json;odata=verbose",
+        },
+      })
+      .catch(() => {
+        return {
+          data: {
+            d: {
+              Name: null,
+            },
+          },
+        };
+      });
+    return res.data.d.Name;
+  }
+
+  /**
+   * A method to get changes from a SharePoint list
+   * @returns - The list of changes
+   */
+  private async getChanges(changeToken: string | null): Promise<ChangeItem[]> {
```

Comment:
**nitpick:** as discussed, maybe optional parameter?

Reply:
done!

Reply:
ah I just remembered why I put null there. If when we initialize, the sp document library is empty, i.e. no changes ever happened. The changeToken is set to `null`, and when the first change occurs. we check against null. 

It made more sense to set it to `null` over `undefined` to cover this case as its not truly a undefined state. 
does that make sense?

Reply:
Yep, makes sense